packages/@aws-cdk/aws-s3-deployment/README.md

```diff
@@ -70,6 +64,37 @@ By default, the contents of the destination bucket will **not** be deleted when
 changed. You can use the option `retainOnDelete: false` to disable this behavior,
 in which case the contents will be deleted.
 
+Configuring this has a few implications you should be aware of:
```

Comment:
I considered dropping this feature entirely due to these, and instead instruct users to use `autoDeleteObjects` on the destination bucket. 

The problem is that `autoDeleteObjects` doesn't respect prefixes (nor it should), so its going to delete the entire bucket, while `retainOnDelete: false` only deletes the specified prefix that it was configured with. 

Since our default is `true`, and we haven't seen any users report these issues, it's probably ok to just surface them a bit and leave as is.